frontend/src/Playlist/Playlist.js

```diff
@@ -0,0 +1,149 @@
+  const [currentplaylist, setplaylist] = useState(0);
+
+  // console.log("now", Object.keys(playlist)[0]);
+  console.log("lop", currentplaylist);
+  let datapass =
+    playlist[currentplaylist] === undefined
+      ? []
+      : playlist[currentplaylist].videos;
+
+  let dataarr = datapass.map((item) => {
+    return data.find((item1) => item1.id === item);
+  });
+
+  let sortedData = getSortedData(dataarr, sortBy);
+  // console.log("sorted", sortedData);
+  let filteredData = getFilteredData(sortedData, { showCategory });
+  // console.log("filtered", filteredData);
+
+  function Showplaylist() {
+    return filteredData.map((item) => {
+      return (
+        <>
+          {" "}
+          <div className="playlist-card">
+            <Link to={`/${item.id}`}>
+              {" "}
+              <div className="playlist-card-data">
+                <img
+                  src={`https://i.ytimg.com/vi/${item.id}/mqdefault.jpg`}
+                  onClick={() => {
+                    dispatch({
+                      type: "ADD_TO_HISTORY",
+                      payload: {
+                        ...item,
+                        ishistory: true,
+                        lastseen: new Date()
+                      }
+                    });
+                  }}
+                  alt=""
+                />
+                <div className="playlist-card-desc">{item.title}</div>
+              </div>
+            </Link>
+            <button
+              class="icon-button lg"
+              onClick={() => {
+                dispatch({
+                  type: "REMOVE_FROM_PLAYLIST",
+                  payload: {
+                    playlistid: playlist[currentplaylist].id,
+                    videoid: item.id
+                  }
+                });
+              }}
+            >
+              <i class="fas fa-times"></i>
+            </button>
+          </div>
+        </>
+      );
+    });
+  }
+  function Removeandupdate(index) {
+    console.log("enter");
+    setplaylist((prev) => {
+      if (prev === index) {
+        if (index + 1 < playlist.length) {
+          console.log("index+1");
+          return index;
+        } else {
+          console.log("index-1");
+          return index - 1;
+        }
+      }
+      console.log("index");
+      if (prev === playlist.length - 1) {
+        return prev - 1;
+      }
+      return prev;
+    });
+  }
+  return (
+    <>
+      <Navbar />
+      <Navigator value="playlist" />
+      <div className="wrapper">
+        <h4>
+          {playlist[currentplaylist] !== undefined
+            ? "Playlist Names"
+            : "No Playlist Present"}
+        </h4>
+        <ul class="list playlistname">
+          {playlist.map((item, index) => {
+            return (
+              <>
+                <li
+                  style={{
+                    background: currentplaylist.id === item.id ? "#f2f2f2" : ""
+                  }}
+                >
```

Comment:
The background color you have given to differentiate the active playlist is not working, state variable "currentplaylist" is of type number and not an object maybe this is the issue.